src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineViewModel_SelectTreeNode.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Collections;
+using Microsoft.VisualStudio.Text.Editor;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.DocumentOutline
+{
+    internal partial class DocumentOutlineViewModel
+    {
+        /// <summary>
+        /// Queue of caret positions that we want to select to corresponding node for.
+        /// </summary>
+        private readonly AsyncBatchingWorkQueue<CaretPosition> _selectTreeNodeQueue;
+
+        /// <summary>
+        /// Last used caret position, read by our model update task so we can keep nodes selected across snap-shot updates.
+        /// </summary>
+        private CaretPosition? _currentlySelectedSymbolCaretPosition;
```

Comment:
i don't quite understand this.  If the snapshot is updated, we can know which needs to be selected since we'll know where the user caret is anyways.   Why not just have that item be preselected when computing the new snapshot?

Reply:
i also only see this being written, not read here.  def confused by it.